solidity/contracts/KeepRandomBeaconOperator.sol

```diff
@@ -218,14 +216,18 @@ contract KeepRandomBeaconOperator is ReentrancyGuard {
 
     /// @notice Adds service contract
     /// @param serviceContract Address of the service contract.
-    function addServiceContract(address serviceContract) public onlyServiceContractUpgrader {
+    function addServiceContract(address serviceContract) public {
+        require(
+            registry.serviceContractUpgraderFor(address(this)) == msg.sender,
+            "Not authorized"
+        );
+
         serviceContracts.push(serviceContract);
     }
 
-    /// @notice Removes service contract
-    /// @param serviceContract Address of the service contract.
-    function removeServiceContract(address serviceContract) public onlyServiceContractUpgrader {
-        serviceContracts.removeAddress(serviceContract);
+    /// @notice Pulls the most recent gas price from gas price oracle.
+    function refreshGasPrice() public {
+        gasPriceCeiling = gasPriceOracle.gasPrice();
     }
```

Comment:
I am still not sure if we should use pull or push model here.

The benefit of the current solution is that we don't need to have an additional require with `msg.sender` and it's a little less developer-error-prone because of that. Also, the contract may decide whether it wants to pull the most recent price at this moment. I think it's similar to `SatWeiPriceFeed` in tBTC and MKR oracle.

On the other hand, `refreshGasprice(uint256 value) onlyGasPriceOracle` may look a little easier and consume less gas. I don't think we can have `onlyGasPriceOracle` modifier given the contract size problems but we probably have enough space to do:
```
    function refreshGasPrice(uint256 newPrice) public {
        require(msg.sender == address(gasPriceOracle), "Not authorized");
        gasPriceCeiling = newPrice;
    }
```

Really interested in reviewer's thoughts! 🙌

Reply:
This seems fine as-is, honestly.

Reply:
>Also, the contract may decide whether it wants to pull the most recent price at this moment.

Wouldn't it be pulled to the operator contract on each update by `finalizeGasPriceUpdate`? What I mean is that the contract will not have a chance to be behind the current oracle's value. Which is fine I guess. Unless we want to have two versions of `finalizeGasPriceUpdate` where we call refresh or not.

I'm fine with the current solution! Just wanted to clarify with y'all that my understanding of how things work is correct.

Reply:
`KeepRandomBeacon` implements `refreshGasPrice` in a way that is pulling the change immediately in the same transaction where `finalizeGasPriceUpdate` is called, that's correct. What I wanted to say is that with the current interface we may have another contract added as the oracle's consumer which - in some circumstances - may decide to not update the price immediately.

Reply:
Well, actually not updating the price immediately would be also possible with push model of `refreshGasPrice(uint256 newPrice)` as long as the new price is public on the oracle. I think the main advantage of the current solution is that there is harder to make a mistake in a consumer contract and let anyone update the price if the appropriate ACL is not in place. With this solution, the change is pulled from the oracle contract so no ACL is needed.